modules/analysis/src/service/load.rs

```diff
@@ -346,6 +347,267 @@ impl InnerService {
             .await
     }
 
+    #[instrument(skip(self, connection), err(level=Level::INFO))]
+    pub(crate) async fn load_latest_graphs_query<C: ConnectionTrait>(
+        &self,
+        connection: &C,
+        query: GraphQuery<'_>,
+    ) -> Result<Vec<(String, Arc<PackageGraph>)>, Error> {
+        let latest_sbom_ids = match query {
+            GraphQuery::Component(ComponentReference::Id(node_id)) => {
+                let sql = r#"
+                  SELECT distinct sbom_id
+                  FROM (
+                      SELECT sbom.sbom_id, sbom.published, product.id, cpe.id,
+                      RANK() OVER (PARTITION BY cpe.id ORDER BY sbom.published DESC) AS rank
+                      FROM sbom_node
+                      LEFT JOIN sbom ON sbom.sbom_id = sbom_node.sbom_id
+                      LEFT JOIN product_version on sbom.sbom_id = product_version.sbom_id
+                      LEFT JOIN product on product.id = product_version.product_id
+                      LEFT JOIN sbom_package_cpe_ref on sbom.sbom_id = sbom_package_cpe_ref.sbom_id
+                      LEFT JOIN cpe ON sbom_package_cpe_ref.cpe_id = cpe.id
```

Comment:
Are all these SELECT's identical except for the WHERE clause? I'd prefer we keep all the redundant parts in a single variable.

Reply:
I was thinking the optimisation here would be to rewrite all those into sea_orm ... how I have no clue ... less enthusiastic of DRY on SQL statements themselves (and attendent string munging).